### PR TITLE
[feat]: [CCM-12733]: Add validations on account in managed account APIs

### PR DESCRIPTION
**Pull request by svmav**
## Describe your changes
1. Added check that the mspAccountId and the managedAccountId are valid
2. Added InternalApi annotation over save/update/delete APIs

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/49157)

### CODE REVIEW

**Comment by svmav** [comment]
trigger uts

**Comment by svmav** [comment]
trigger uts

File: 490-ce-commons/src/main/java/io/harness/ccm/msp/service/impl/ManagedAccountServiceImpl.java [hunk 1]
```diff
@@ -7,19 +7,37 @@
...
 
 import com.google.inject.Inject;
 import java.util.List;
 
 public class ManagedAccountServiceImpl implements ManagedAccountService {
   @Inject private ManagedAccountDao managedAccountDao;
+  @Inject private AccountClient accountClient;
+
+  private static final String INVALID_ACCOUNT_ID_EXCEPTION =
+      "Error is performing operation. Invalid mspAccountId/managedAccountId.";
```
Comment (akash-nagarajan) [inline 490-ce-commons/src/main/java/io/harness/ccm/msp/service/impl/ManagedAccountServiceImpl.java:26]: We dont need "Error is performing operation". We should throw 400 status with "Invalid account id" / The account is not a managed service provider account based on which account is invalid

File: 490-ce-commons/src/main/java/io/harness/ccm/msp/service/impl/ManagedAccountServiceImpl.java [hunk 2]
```diff
@@ -7,19 +7,37 @@
...
 
   @Override
-  public String save(ManagedAccount managedAccount) {
-    return managedAccountDao.save(managedAccount);
+  public String save(String mspAccountId, String managedAccountId) {
+    AccountDTO mspAccountDTO = CGRestUtils.getResponse(accountClient.getAccountDTO(mspAccountId));
```
[the remark this answers is not in the document]
Comment (svmav) [reply 490-ce-commons/src/main/java/io/harness/ccm/msp/service/impl/ManagedAccountServiceImpl.java:31]: This check will be added once the Account DTO changes related to MSP are merged and deployed.

File: 490-ce-commons/src/main/java/io/harness/ccm/msp/service/impl/ManagedAccountServiceImpl.java (hunk 1 above)
Comment (svmav) [reply 490-ce-commons/src/main/java/io/harness/ccm/msp/service/impl/ManagedAccountServiceImpl.java]: Done

**Comment by svmav** [comment]
trigger uts

**Review by akash-nagarajan — APPROVED** [review]

**Comment by svmav** [comment]
trigger uts

**Comment by svmav** [comment]
trigger uts

**Pull request merged**